base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -89,24 +89,27 @@ public void solveMetas() {
       case Expr.Lambda(var ref, var body) -> switch (whnf(type)) {
         case DepTypeTerm(var kind, var dom, var cod) when kind == DTKind.Pi -> {
           // unifyTyReported(param, dom, expr);
-          var core = subscoped(ref, dom, () ->
-            inherit(body, cod.apply(new FreeTerm(ref))).wellTyped()).bind(ref);
-          yield new Jdg.Default(new LamTerm(core), type);
+          try (var ignored = subscope(ref, dom)) {
+            var core = inherit(body, cod.apply(new FreeTerm(ref))).wellTyped().bind(ref);
+            yield new Jdg.Default(new LamTerm(core), type);
+          }
         }
         case EqTerm eq -> {
-          var core = subscoped(ref, DimTyTerm.INSTANCE, () ->
-            inherit(body, eq.appA(new FreeTerm(ref))).wellTyped()).bind(ref);
-          checkBoundaries(eq, core, body.sourcePos(), msg ->
-            new CubicalError.BoundaryDisagree(expr, msg, new UnifyInfo(state)));
-          yield new Jdg.Default(new LamTerm(core), eq);
+          try (var ignored = subscope(ref, DimTyTerm.INSTANCE)) {
+            var core = inherit(body, eq.appA(new FreeTerm(ref))).wellTyped().bind(ref);
+            checkBoundaries(eq, core, body.sourcePos(), msg ->
+              new CubicalError.BoundaryDisagree(expr, msg, new UnifyInfo(state)));
+            yield new Jdg.Default(new LamTerm(core), eq);
```

Comment:
Can you move these two lines to the outside?

Reply:
done